### PR TITLE
Fix route delete bug

### PR DESCRIPTION
**Pull request by ghost**
image

### CODE REVIEW

**Review by francoisferrand — APPROVED** [review]

File: lib/s3routes/routes/routeDELETE.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default function routeDELETE(
                  */
                 if (err && (
                     !(err instanceof ArsenalError) || 
-                    (!err.is.NoSuchKey && err.is.NoSuchVersion)
+                    (!err.is.NoSuchKey && !err.is.NoSuchVersion)
```
Comment (francoisferrand) [inline lib/s3routes/routes/routeDELETE.ts:63]: Can we add unit tests for these functions?

**Review by francoisferrand — APPROVED** [review]

**Review by benzekrimaha — APPROVED** [review]

File: lib/s3routes/routes/routeDELETE.ts (hunk 1 above)
Comment (ghost) [reply lib/s3routes/routes/routeDELETE.ts:63]: I added some but they don't cover all the paths, I will add it (while I am confirming in cloudserver the fix)

Comment (ghost) [reply lib/s3routes/routes/routeDELETE.ts:63]: (we had 0 unit test for the routes before)

Comment (ghost) [reply lib/s3routes/routes/routeDELETE.ts:63]: Done in https://github.com/scality/Arsenal/pull/2273/commits/24a87f59fd4959bfebdc94bf0b32fb1d86bd0b27

we already have it but it was not testing both cases...

**Pull request merged**